pkg/chunk/encoding/bigchunk.go

```diff
@@ -169,13 +171,19 @@ func (b *bigchunk) Size() int {
 	return sum
 }
 
-func (b *bigchunk) NewIterator() Iterator {
+func (b *bigchunk) NewIterator(reuseIter Iterator) Iterator {
 	var it chunkenc.Iterator
 	if len(b.chunks) > 0 {
-		it = b.chunks[0].Iterator(nil)
+		it = b.chunks[0].Iterator(it)
```

Comment:
This should pass in `bci.curr`, so we get reuse at the lower level too.

Reply:
Nice catch! Fixed it.